lib/cocoapods/command/repo/add.rb

```diff
@@ -83,8 +85,25 @@ def clone_repo
             Dir.chdir(config.repos_dir) do
               command = ['clone', @url]
               command << '--progress' if @progress
+              command << %w[--filter=blob:none --no-checkout --depth 1] if @sparse
               command << '--' << @name
+
               git!(command)
+
+              if @sparse
+                Dir.chdir(@name) do
+                  git!(%w[config core.sparseCheckout true])
+                  git!(%w[config core.sparseCheckoutCone true])
+                  File.open('.git/info/sparse-checkout', 'w') do |sparse|
+                    sparse.puts '/*'
+                    sparse.puts '!/*/'
+                    sparse.puts 'Specs/2/e/c/RxSwift'
```

Comment:
what is this needed for?

Reply:
Well, this is a funny thing, actually - the `Specs` dir has no direct files, only folders - so using the 2 filters above, git doesn't create it together with the rest of the repo root.   
Because there is no `Specs` dir at the time of creation, `Source#specs_dir` returns an incorrect value.

Adding this arbitrary pod dir causes `Specs` to get created and therefore the rest of the process to work correctly.

We could add a `.gitkeep` file into the spec repo and have a less esoteric sparse checkout init, what do you think?

Reply:
Adding a `.gitkeep` or similar (preferably with content that explains its use) would be preferred to an arbitrary pod I think